sound/soc/intel/boards/sof_sdw.c

```diff
@@ -253,18 +253,25 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 	},
 };
 
-static inline int find_codec_info_part(unsigned int part_id)
+static inline int find_codec_info_part(u64 adr)
 {
+	unsigned int part_id, sdw_version;
 	int i;
 
+	part_id = SDW_PART_ID(adr);
+	sdw_version = SDW_VERSION(adr);
 	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
-		if (part_id == codec_info_list[i].id)
-			break;
+		/*
+		 * A codec info is for all sdw version with the part id if
+		 * vendor_id is not specified in the codec info.
+		 */
+		if (part_id == codec_info_list[i].id &&
+		    (!codec_info_list[i].vendor_id ||
+		     sdw_version == codec_info_list[i].vendor_id))
```

Comment:
it just looks a bit strange for someone, who hasn't read the SDW spec: the "vendor ID" is really the SDW version? Where does this come from? Maybe we should use the same name then or at least add a bit about that to the comment?

Reply:
> it just looks a bit strange for someone, who hasn't read the SDW spec: the "vendor ID" is really the SDW version? Where does this come from? Maybe we should use the same name then or at least add a bit about that to the comment?

Thanks for pointing this out. I think vendor_id is a typo. Rename it to version_id now.